### PR TITLE
update swagger API doc from jupyter-js-services

### PR DESCRIPTION
**Pull request by minrk**

### CODE REVIEW

**Comment by blink1073** [comment]
Is this an exact copy of what was in `jupyter-js-services`?

**Comment by minrk** [comment]
@blink1073 yes, other than bumping the version to 5.

**Comment by minrk** [comment]
@blink1073 actually, after checking it with swagger, there were some validation errors that I've now fixed.

**Comment by blink1073** [comment]
LGTM

File: notebook/services/api/api.yaml [hunk 1]
```diff
@@ -27,8 +27,315 @@ parameters:
...
+    post:
+      summary: Restore a file to a particular checkpointed state
+      parameters:
+        - $ref: "#/parameters/path"
+        - $ref: "#/parameters/checkpoint_id"
+      tags:
+        - contents
+      responses:
+        204:
+          description: Checkpoint created
```
Comment (jasongrout) [inline notebook/services/api/api.yaml]: This description should be something like "Checkpoint restored", not created.

**Comment by jasongrout** [comment]
One small comment, but other than that, looks good to me.

**Comment by minrk** [comment]
@jasongrout thanks! fixed.

**Comment by jasongrout** [comment]
I'm trying to check the rendering, and for some reason it isn't rendering: http://petstore.swagger.io/?url=https://raw.githubusercontent.com/minrk/notebook/swagger-doc/notebook/services/api/api.yaml

**Comment by jasongrout** [comment]
It looks like a bug in the swagger renderer.  I'm looking into it.

**Comment by jasongrout** [comment]
Yep, definitely looks like a bug in swagger: https://github.com/swagger-api/swagger-ui/issues/1903

**Comment by jasongrout** [comment]
How do I kickstart Travis again?

**Comment by blink1073** [comment]
Does hitting the restart button here not work for you?  https://travis-ci.org/jupyter/notebook/jobs/103882148

screen shot 2016-01-21 at 11 58 29 am

**Comment by jasongrout** [comment]
I hit it several times, and finally all of them passed.

**Comment by willingc** [comment]
@jasongrout, @blink1073, @minrk Do we want to link to the swagger API docs from the Developer section of the Sphinx docs or README.md?

**Comment by blink1073** [comment]
I'd say so (to both).

**Comment by willingc** [comment]
Ping me when the rendering bug for swagger is solved, send me a link to where it's rendered, and I'm happy to update both.

**Comment by jasongrout** [comment]
Yeah, I'd say that we figure out how to statically render that page and include it in the notebook as a static file.  It's silly that we are depending on a hosted service to do the static rendering.

**Comment by jasongrout** [comment]
and include it in the hosted documentation as well...

**Comment by blink1073** [comment]
https://www.npmjs.com/package/bootprint-swagger

**Pull request merged**